WordPress/gutenberg
# Site and Post Editor: Unify the DocumentBar component

Related #52632 

## What?

This PR updates the post and site editors, to use the exact same `DocumentBar` component and replaces both `DocumentActions` component that are visible in the site and post editors.

To be able to do so, this PR had to switch the "template mode" of the post editor to only show the "template" without the edited post and page which matches the behavior of the site editor. (as discussed in the issue)

There are still some differences between how the modes work that I didn't update in this PR. 

 - Editing a page in the post editor lands in "post-only" mode (means we only show the title and content by default)
 - Editing a page in the site editor uses the "template-locked" mode where the template is visible but locked.
 - The document bar is always shown in the site editor while in the post editor it's only shown in "template mode". (not entirely sure why but kept it as is)

I left this consideration separate.

I also introduced a new `defaultRenderingMode` setting to the `EditorProvider` to define the default mode that the editor needs to be in when initially rendering the current post/page/template...

## Testing Instructions

1- Check the template mode in the post editor, notice that it only shows the template now.
2- Check the document bar is shown properly while editing different entities in post and site editors.
